os/arch/arm/src/armv7-m/up_blocktask.c

```diff
@@ -114,6 +114,10 @@ void up_block_task(struct tcb_s *tcb, tstate_t task_state)
 
 	ASSERT((tcb->task_state >= FIRST_READY_TO_RUN_STATE) && (tcb->task_state <= LAST_READY_TO_RUN_STATE));
 
+#ifdef CONFIG_TASK_SCHED_HISTORY
+	/* Save the task name which has to be blocked */
+	save_task_scheduling_status(tcb);
```

Comment:
As per my understanding, this is used to log the context switching for each task. 
If you want to log when the task is switched out, then the above change is good. But, this function must not be called at line 180.